allennlp/modules/text_field_embedders/model_text_field_embedder.py

```diff
@@ -0,0 +1,72 @@
+from typing import Dict
+
+import torch
+from overrides import overrides
+
+from allennlp.common import Params
+from allennlp.data import Vocabulary
+from allennlp.models.model import Model
+from allennlp.modules.text_field_embedders.text_field_embedder import TextFieldEmbedder
+
+
+@TextFieldEmbedder.register("model")
+class ModelTextFieldEmbedder(TextFieldEmbedder):
+    """
+    This is a ``TextFieldEmbedder`` that wraps a ``Model``.  The model must accept a single text
+    field as input.  We run ``forward`` on the model, then pull out one of the values in its output
+    dictionary to return.
+
+    NOTE: This is still largely experimental and needs some real-world testing before I'm confident
+    this API is correct and actually works well.
+
+    Parameters
+    ----------
+    model : ``Model``
+        The model that we will use to embed the text field.
+    model_input_name : ``str``
+        The name the model assigns to the text field input we want to use.  For example, the
+        :class:`~allennlp.models.SimpleTagger` model takes a text input field named ``tokens`` -
+        that's the name that should be passed here.
+    model_output_name : ``str``
+        The key that we will look for in the model's output dictionary.  The model must have
+        included a token-level representation in its output dictionary for it to be useful here.
+    model_output_dim : ``int``
+        ``TextFieldEmbedders`` need to implement ``get_output_dim()``.  We could either make the
+        model implement some method that tells us what the output dim is, or we can take it as a
+        parameter.  We're opting for the second option here.
+    """
+    def __init__(self,
+                 model: Model,
+                 model_input_name: str,
+                 model_output_name: str,
+                 model_output_dim: int) -> None:
+        super(ModelTextFieldEmbedder, self).__init__()
+        self._model = model
+        self._input_name = model_input_name
+        self._output_name = model_output_name
+        self._output_dim = model_output_dim
+
+    @overrides
+    def get_output_dim(self) -> int:
+        return self._output_dim
+
+    def forward(self, text_field_input: Dict[str, torch.Tensor]) -> torch.Tensor:
```

Comment:
This is overly restrictive. In order to get embeddings out of the SRL model, we also need to pass in a `IndexField`. I think this can just be `kwargs` or something which gets passed down to the Model?